docs/DeveloperGuide.md

```diff
@@ -365,6 +392,49 @@ The sequence diagram of `updateClient`, `updateProperty` and `updatePair` can be
 Note that when delete operation is being invoked on client and property, the `updatePair` method will also be invoked to
 prevent entries retaining within pairingList after it has been deleted from clientList or propertyList.
 
+
+### List feature
+
+There are 3 main steps whenever a list command needs to be executed
+* When the user enters any command, it first needs be understood. That is handled by the ParseManager class.
+   Next, when the first word entered by the user is determined to be `list`, ParseManager itself then determines
+   the type of list command entered, including the tags. ParseListClient, ParseListProperty and ParseListEverything
+   are checkers to ensure that a valid command has been entered. ParseListClient and ParseListProperty also determine
+   if tags have been entered, and if those tags are valid.
+  ```
+        case COMMAND_LIST:
+            ArrayList<String> listCommandTypeAndFlags = getListCommandType(commandDetail);
+            boolean isListProperty = listCommandTypeAndFlags.get(0).trim().equals(PROPERTY_FLAG);
+            boolean isListClient = listCommandTypeAndFlags.get(0).equals(CLIENT_FLAG);
+            boolean isListEverything = listCommandTypeAndFlags.get(0).equals(EVERYTHING_FLAG);
+            if (isListProperty) {
+                return new ParseListProperty(listCommandTypeAndFlags.get(1));
+            } else if (isListClient) {
+                return new ParseListClient(listCommandTypeAndFlags.get(1));
+            } else if (isListEverything && listCommandTypeAndFlags.get(1).isEmpty()) {
+                return new ParseListEverything();
+            } else {
+                throw new UndefinedSubCommandTypeException(MESSAGE_INCORRECT_LIST_DETAILS);
+            }
+  ```
+This block of code is part of ParseManager. It determines the type of list operation(-client, 
+-property or -everything) and returns the corresponding object.  
+Both ParseListClient and ParseListObject then determine if tags are present, and if they are valid, 
+throwing exceptions if any errors are encountered. They then return the corresponding Command type necessary.
+
+* There are five different classes which handle each of the features described above. Each class inherits from the
+   abstract Command class, and reads information present in either the PropertyList or ClientList objects respectively.
+
+The execute function retrieves the propertyList holding all the current properties. It then loops 
+through every property present. For every property, it passes it to the corresponding display function
+in Ui.
```

Comment:
Can consider rephrasing this paragraph as well. The first sentence is not very coherent. For the last sentence, the part on "it passes it" can be rephrase.